### ISSUE TITLE
Resources don't seem to be adding resource name when created

### ISSUE THREAD

**Issue by jaxxstorm** [pulumi/pulumi-oci#55]
### What happened?

Creating OCI resources doesn't appear to be adding the pulumi resource name

### Steps to reproduce

Create any resource, for example a VCN
```
vcn = new oci.core.Vcn(
      "example",
      {
        compartmentId: <someid>,
        cidrBlocks: [<someCidr>],
        
      },
  );
```

### Expected Behavior

I would expect the resulting vcn to be called something like `example-<randomstring>`

### Actual Behavior

It creates a vcn with a set name:

```
vcn<randomstring>
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by lukehoban** [comment]
This resource doesn’t have a `name` property so doesn’t get auto-naming applied by default. 

It’s possible that the `displayName` property should be autonamed?  (Though if it’s truly a display name, also doesn’t seem parricularly necessary to autoname it?)

@jaxxstorm which property did you expect to be autonamed?

**Comment by jaxxstorm** [comment]
I think you're right, `displayName` seems to be the property I expected to have the name appended.

If you're creating something in pulumi called `example` I'd expect to be able to go into the console and find it based on that name. However currently, just uses the display name `<resource>`

**Comment by guineveresaenger** [comment]
I can fix this; we need to set `displayName` to the autonaming field explicitly on this resource. Example: https://github.com/pulumi/pulumi-confluentcloud/blob/main/provider/resources.go#L82

**Comment by guineveresaenger** [comment]
I missed earlier that it appears this entire provider opts to use `displayName` for resources where others use `name`. That's a much more involved fix.

@jaxxstorm it looks like we're hardcoding this here: https://github.com/pulumi/pulumi-terraform-bridge/blob/master/pkg/tfbridge/provider.go#L1356

The override functionality lives here: https://github.com/pulumi/pulumi-terraform-bridge/blob/master/pkg/tfbridge/names.go#L147

I wonder if we should allow `displayName` as an option for the `SetAutoNaming` function, since we should probably not assume a blanket `name` field. If so, could you open an issue for us in pulumi-terraform-bridge @jaxxstorm ?

**Comment by gpoul** [comment]
I don't think you can use this as-is, but I tested a VCN creation with autonaming using the changes in [my branch oci-autonaming](https://github.com/gpoul/pulumi-oci/tree/oci-autonaming).

Would be awesome if we could get autonaming working with the OCI provider.
